crates/cheatcodes/spec/src/vm.rs

```diff
@@ -306,6 +306,10 @@ interface Vm {
     #[cheatcode(group = Evm, safety = Safe)]
     function signP256(uint256 privateKey, bytes32 digest) external pure returns (bytes32 r, bytes32 s);
 
+
+    // Generate / derive the public key coordinates based on the secp256r1 curve from a random uint256 private key
+    #[cheatcode(group =  Evm , safety = Safe)]
+    function generate_public_key_p256(uint256 privateKey) external pure returns (uint256 publicKeyX, uint256 publicKeyY);
```

Comment:
I think we can rename this to `generateP256` and the return parameters being `x`, `y`

Reply:
it's also not really "generate", public keys are derived from the private keys not generated out of nowhere